### PR TITLE
Sendmail rules: Extend filter by order status

### PR DESCRIPTION
**Pull request by ser8phin**
- [x] Adjust recipient filters
- [x] Adjust API
- [x] Adjust Tests
    - [x] Test 1/2
    - [x] Test 2/2
- [x] Adjust Docs
- [x] Check
- [x] Increase tests / code coverage

### CODE REVIEW

File: doc/api/resources/sendmail_rules.rst [hunk 1]
```diff
@@ -18,8 +18,13 @@ subject                               multi-lingual string       The subject of
...
 limit_products                        list of integers           List of product IDs, if ``all_products`` is not set
-include_pending                       boolean                    If ``true``, the email is sent to pending orders. If ``false``,
+[**DEPRECATED**] include_pending      boolean                    If ``true``, the email is sent to pending orders. If ``false``,
                                                                  only paid orders are considered.
+restrict_to_status                    list                       List of order states to restrict recipients to. Valid
+                                                                 entries are ``p, e, c, pa, na, valid_if_pending`` for
+                                                                 the order states paid, expired, canceled, approval pending,
+                                                                 payment pending, and payment pending but already confirmed.
+                                                                 The default is ``["p", "na", "valid_if_pending"]``.
```
[the remark this answers is not in the document]
Comment (ser8phin) [reply doc/api/resources/sendmail_rules.rst]: No reason I could remember, so changed it

File: src/pretix/plugins/sendmail/migrations/0005_rule_migrate_statusfilter.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('sendmail', '0004_rule_restrict_to_status'),
+        ('pretixbase', '0241_itemmetaproperties_required_values'),
+    ]
+
+    operations = [
```
[the remark this answers is not in the document]
Comment (ser8phin) [reply src/pretix/plugins/sendmail/migrations/0005_rule_migrate_statusfilter.py]: Seems to work :)

File: src/tests/plugins/sendmail/test_rules.py [hunk 3]
```diff
@@ -276,12 +276,120 @@ def test_sendmail_rule_send_correct_products(event, order, item, item2):
...
+    run_restriction_test(event, order, restrictions_pass, restrictions_fail)
+
+
+@pytest.mark.django_db
+@scopes_disabled()
+def test_sendmail_rule_restrictions_status_overdue_pending(event, order):
+    order.status = Order.STATUS_PENDING
+    order.require_approval = True
+    order.valid_if_pending = False
+    order.expires = order.datetime - datetime.timedelta(hours=1)  # TODO: make this test-order actually overdue
```
Comment (ser8phin) [inline src/tests/plugins/sendmail/test_rules.py:356]: This still needs more thoughts, as either the overdue check is wrong or the actual overdue restriction

Comment (raphaelm) [reply src/tests/plugins/sendmail/test_rules.py:356]: I think overdue might only be a valid choice if

event.settings.payment_term_expire_automatically = True

is set first?

File: src/tests/api/test_sendmail.py [hunk 4]
```diff
@@ -129,6 +129,9 @@ def test_sendmail_rule_create_minimal(token_client, organizer, event):
             'subject': {'en': 'meow'},
             'template': {'en': 'creative text here'},
             'send_date': '2018-03-17T13:31Z',
+            "restrict_to_status": ['p', 'valid_if_pending'],
+            # maybe this (the default) should be set when not giving any options?
+            # but then it's not entirely clear elsewhere…
```
Comment (raphaelm) [inline src/tests/api/test_sendmail.py:134]: What do you mean exactly? Yes, if the field is missing, the default must be used.

Is this then a TODO that should be resolved merging?

File: src/tests/api/test_sendmail.py [hunk 5]
```diff
@@ -253,6 +256,59 @@ def test_sendmail_rule_create_invalid(token_client, organizer, event):
...
+
+    create_rule(
+        token_client, organizer, event,
+        data={
+            'subject': {'en': 'meow'},
+            'template': {'en': 'creative text here'},
+            'send_date': '2018-03-17T13:31Z',
+        },
+        expected_failure=True,
+        expected_failure_text="restrict_to_status needs at least one value"
```
Comment (raphaelm) [inline src/tests/api/test_sendmail.py:308]: See above, this should use the default, otherwise the API is not backwards-compatible

Comment (raphaelm) [reply src/tests/api/test_sendmail.py:308]: Also, what happens if someone sends `include_pending` in there API request, are we "translating" that?

Comment (ser8phin) [reply src/tests/api/test_sendmail.py]: We are doing that now :)

File: doc/api/resources/sendmail_rules.rst [hunk 6]
```diff
@@ -37,7 +44,10 @@ send_to                               string                     Can be ``"order
                                                                  or ``"both"``.
                                                                  date. Otherwise it is relative to the event start date.
 ===================================== ========================== =======================================================
+.. versionchanged:: 5.0
```
Comment (raphaelm) [inline doc/api/resources/sendmail_rules.rst:47]:
```suggestion
.. versionchanged:: 2023.7
```

**Review by raphaelm — APPROVED** [review]

**Pull request merged**